src/string-calculator.ts

```diff
@@ -1,12 +1,205 @@
+                    let position:number = this.getIndexOfUnexpectedCustomSeparator(numberList);
+                    if(errorMessage.trim().length!=0) {
+                        errorMessage=errorMessage.concat('\n');
+                    }
+                    errorMessage=errorMessage.concat(`Number expected but \'${customSep}\' found at position ${position}.`);
+                    lastError = new CustomSeparatorFoundException;
+
+                } else if(Error instanceof EndOfLineFoundException) {
+                    let position:number = this.getIndexOfUnexpectedEndOfLine(numberList);
+                    if(errorMessage.trim().length!=0) {
+                        errorMessage=errorMessage.concat('\n');
+                    }
+                    errorMessage=errorMessage.concat(`Number expected but \'\\n\' found at position ${position}.`);
+                    lastError = new LastNumberMissedException;
+
+                } else if(Error instanceof LastNumberMissedException) {
+                    if(!isUnexpectedEOLAdded) {
+                        if(errorMessage.trim().length!=0) {
+                            errorMessage=errorMessage.concat('\n');
+                        }
+                        errorMessage=errorMessage.concat(`Number expected but EOF found.`);
+                        isUnexpectedEOLAdded = true;
+                    }
+                    lastError = new LastNumberMissedException;
+
+                } else if(Error instanceof NegativeNumberException) {
+                    if(lastError instanceof NegativeNumberException) {
+                        errorMessage=errorMessage.concat(", "+number);
+                    } else {
+                        if(errorMessage.trim().length!=0) {
+                            errorMessage=errorMessage.concat('\n');
+                        errorMessage=errorMessage.concat(`Negative not allowed : `);
+                        }
+                    }
+
+                    lastError = new NegativeNumberException;
+                } else if(Error instanceof NumberNotNumericException) {
+                    let unexpectedChar = this.getUnexpectedChar(number);
+                    let position = numberList.indexOf(unexpectedChar);
+
+                    if(errorMessage.trim().length!=0) {
+                        errorMessage=errorMessage.concat('\n');
+                    }
+                    errorMessage=errorMessage.concat(`'${customSep}' expected but '${unexpectedChar}' found at position ${position}.`)
+
+                    lastError = new NumberNotNumericException;
+                }
+            }
+        });
+
+
+        if(errorMessage.trim().length!=0) {
+            return errorMessage;
+        }
+        return result.toString();
+    }
+
+    private createRegex(numberList: string): string {
+        let res = ',|\n';
+        if(numberList.includes("//")){
+            let sep = this.getCustomSeparator(numberList);
+            if(
```

Comment:
i'm sure these escape rules are common so there must be something or someone that has already written this logic